libs/Matchmaking/src/IDiscoveryAgent.cs

```diff
@@ -11,6 +11,14 @@ namespace Microsoft.MixedReality.Sharing.Matchmaking
     /// <summary>
     /// Handle to an ongoing discovery subscription.
     /// </summary>
+    /// <remarks>
+    /// A subscription is bound to the originating <see cref="IDiscoveryAgent"/>, but its methods/properties
+    /// (including `Dispose()`) must be safe to use independently from the state of the agent - even from a
+    /// different thread from the one where the agent is used.
+    ///
+    /// After the agent is disposed, the result of <see cref="Resources"/> becomes undefined, and
+    /// <see cref="Updated"/> might stop being raised.
```

Comment:
This is a bit confusing. So, each call to Resources returns you some enumerable. I'd expect that it updates dynamically, and doesn't really reflect the most recent state of things (just due to MT issues at least).

Why do we have to UB this instead of, let's say, saying that eventually we'll start returning an empty enumerable, or throwing an exception? Is there some difficult synchronization logic that makes this a requirement?

Reply:
Hmm not really. Returning an empty enumerable sounds like a reasonable behavior.

Reply:
Fixed